staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/leaselock.go

```diff
@@ -0,0 +1,181 @@
+func (ll *LeaseLock) Update(ler LeaderElectionRecord) error {
+	if ll.l == nil {
+		return errors.New("lease not initialized, call get or create first")
+	}
+	spec, err := LeaderElectionRecordToLeaseSpec(&ler)
+	if err != nil {
+		return err
+	}
+	ll.l.Spec = spec
+	ll.l, err = ll.Client.Leases(ll.LeaseMeta.Namespace).Update(ll.l)
+	return err
+}
+
+// RecordEvent in leader election while adding meta-data
+func (ll *LeaseLock) RecordEvent(s string) {
+	events := fmt.Sprintf("%v %v", ll.LockConfig.Identity, s)
+	ll.LockConfig.EventRecorder.Eventf(&v1beta1.Lease{ObjectMeta: ll.l.ObjectMeta}, v1.EventTypeNormal, "LeaderElection", events)
+}
+
+// Describe is used to convert details on current resource lock
+// into a string
+func (ll *LeaseLock) Describe() string {
+	return fmt.Sprintf("%v/%v", ll.LeaseMeta.Namespace, ll.LeaseMeta.Name)
+}
+
+// returns the Identity of the lock
+func (ll *LeaseLock) Identity() string {
+	return ll.LockConfig.Identity
+}
+
+func leaseSpecToLeaderElectionRecord(spec *v1beta1.LeaseSpec) (LeaderElectionRecord, error) {
+	holderIdentity := ""
+	if spec.HolderIdentity != nil {
+		holderIdentity = *spec.HolderIdentity
+	}
+	leaseDurationSeconds := 0
+	if spec.LeaseDurationSeconds != nil {
+		leaseDurationSeconds = int(*spec.LeaseDurationSeconds)
+	}
+	acquireTime, err := microTimeToTime(spec.AcquireTime)
+	if err != nil {
+		return LeaderElectionRecord{}, err
+	}
+	renewTime, err := microTimeToTime(spec.RenewTime)
+	if err != nil {
+		return LeaderElectionRecord{}, err
+	}
+	leaseTransitions := 0
+	if spec.LeaseTransitions != nil {
+		leaseTransitions = int(*spec.LeaseTransitions)
+	}
+	return LeaderElectionRecord{
+		HolderIdentity:       holderIdentity,
+		LeaseDurationSeconds: leaseDurationSeconds,
+		AcquireTime:          acquireTime,
+		RenewTime:            renewTime,
+		LeaderTransitions:    leaseTransitions,
+	}, nil
+}
+
+func LeaderElectionRecordToLeaseSpec(ler *LeaderElectionRecord) (v1beta1.LeaseSpec, error) {
+	leaseDurationSeconds := int32(ler.LeaseDurationSeconds)
+	acquireTime, err := timeToMicroTime(&ler.AcquireTime)
+	if err != nil {
+		return v1beta1.LeaseSpec{}, err
+	}
+	renewTime, err := timeToMicroTime(&ler.RenewTime)
+	if err != nil {
+		return v1beta1.LeaseSpec{}, err
+	}
+	leaseTransitions := int32(ler.LeaderTransitions)
+	return v1beta1.LeaseSpec{
+		HolderIdentity:       &ler.HolderIdentity,
+		LeaseDurationSeconds: &leaseDurationSeconds,
+		AcquireTime:          &acquireTime,
+		RenewTime:            &renewTime,
+		LeaseTransitions:     &leaseTransitions,
+	}, nil
+}
+
+func microTimeToTime(u *metav1.MicroTime) (metav1.Time, error) {
+	if u == nil {
+		return metav1.Time{}, nil
+	}
+	b, err := u.MarshalBinary()
+	if err != nil {
+		return metav1.Time{}, err
+	}
+	t := time.Time{}
+	err = t.UnmarshalBinary(b)
+	if err != nil {
+		return metav1.Time{}, err
+	}
+	return metav1.NewTime(t), nil
+}
+
+func timeToMicroTime(u *metav1.Time) (metav1.MicroTime, error) {
```

Comment:
[https://github.com/kubernetes/kubernetes/blob/16d099253466eb18fddcab9f9adf6214fe4077ad/staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go#L40-L41](https://github.com/kubernetes/kubernetes/blob/16d099253466eb18fddcab9f9adf6214fe4077ad/staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go#L40-L41)

can we switch them to micro time? @mikedanese

Reply:
Why is this helpful?

Reply:
Hmm, I see that LeaseSpec went with microtime. It should be fine. We don't actually use it to implement correctness. Also metav1.MicroTime to metav1.Time is just:
```
time := metav1.Time{microTime.Time}
```

No need for this function.

Reply:
wow :+1: fixed in new patch